iscsi/iscsiadm.go

```diff
@@ -20,60 +23,65 @@ type Secrets struct {
 	PasswordIn string `json:"passwordIn,omitempty"`
 }
 
-func iscsiCmd(args ...string) (string, error) {
-	stdout, err := execWithTimeout("iscsiadm", args, time.Second*3)
+func iscsiCmd(ctx context.Context, args ...string) (string, error) {
+	logger := klog.FromContext(ctx)
+	stdout, err := execWithTimeout(ctx, "iscsiadm", args, time.Second*3)
 
-	debug.Printf("Run iscsiadm command: %s", strings.Join(append([]string{"iscsiadm"}, args...), " "))
-	iscsiadmDebug(string(stdout), err)
+	logger.V(1).Info("Run iscsiadm", "command", strings.Join(append([]string{"iscsiadm"}, args...), " "))
+	iscsiadmDebug(ctx, string(stdout), err)
 
 	return string(stdout), err
 }
 
-func iscsiadmDebug(output string, cmdError error) {
+func iscsiadmDebug(ctx context.Context, output string, cmdError error) {
```

Comment:
So, the majority of functions in this library, if not all, do not require a context. Do you feel that the best practice would be to pass a logger to the library functions and NOT to pass in a context? I was torn on the right call, passing a logger vs context.

Reply:
For the public API definitely use a context because it is more versatile.

Internally use a logger if it looks very likely that a context will never be needed. That can still be changed later on if the estimate turns out to be false.

Reply:
Submitted new changes that use a context for External calls and pass a logger pointer for internal calls.